src/vvtbi.cc

```diff
@@ -0,0 +1,729 @@
+        case Tokenizer::TokenType::LETTER:
+            DEBUG_LOG("Found assignment statement");
+            let_statement();
+            break;
+        default:
+            DEBUG_LOG(
+              "Unrecognized statement type: " << get_token_string(token));
+            dprintf("Syntax Error: Unrecognized statement", E_ERROR);
+            break;
+    }
+}
+
+/**
+ * Parses and executes a line statement.
+ * Handles line numbers and statement execution.
+ */
+void VVTBI::line_statement() {
+    DEBUG_LOG("Starting line_statement");
+    // Skip empty lines
+    while (tokenizer_->current_token() == Tokenizer::TokenType::EOL) {
+        tokenizer_->next_token();
+    }
+    // Check for end of file
+    if (tokenizer_->current_token() == Tokenizer::TokenType::EOF_TOKEN) {
+        execution_finished_ = true;
+        return;
+    }
+    // Process line number if present
+    if (tokenizer_->current_token() == Tokenizer::TokenType::NUMBER) {
+#ifdef DEBUG_MODE
+        int line_num = tokenizer_->get_num();
+        DEBUG_LOG("Processing line number: " << line_num);
+#else
+        tokenizer_->get_num(); // Consume line number
+#endif
+        tokenizer_->next_token();
+    }
+    statement();
+}
+
+/**
+ * Builds a map of line numbers in the program.
+ * Maps each line number to its position in the source.
+ */
+void VVTBI::build_line_map() {
+    DEBUG_LOG("Building line number map");
+    line_positions_.clear();
+    tokenizer_->reset();
+    std::unordered_map<int, bool> found_lines;
+    // Scan through tokens looking for line numbers
+    while (!tokenizer_->finished()) {
+        auto token = tokenizer_->current_token();
+        DEBUG_LOG("Map building - current token: " << get_token_string(token));
+
+        if (token == Tokenizer::TokenType::NUMBER) {
+            int value = tokenizer_->get_num();
+            if (is_valid_line_number(value)) {
+                line_positions_[value] = true;
+                found_lines[value] = true;
+                DEBUG_LOG("Found line number: " << value);
+            }
+        }
+        tokenizer_->next_token();
+    }
+#ifdef DEBUG_MODE
+    log_found_line_numbers(found_lines);
+#endif
+    tokenizer_->reset();
+}
+
+#ifdef DEBUG_MODE
+/**
+ * Helper to log all found line numbers during map building
+ */
+void VVTBI::log_found_line_numbers(
+  const std::unordered_map<int, bool>& found_lines) {
+    std::string line_numbers;
+    for (const auto& [line, _] : found_lines) {
+        if (!line_numbers.empty()) {
+            line_numbers += " ";
+        }
+        line_numbers += std::to_string(line);
+    }
+    DEBUG_LOG("Found these line numbers: " << line_numbers);
+}
+#endif
+/**
+ * Finds a specific line number in the source code.
+ * Builds line map if not already built.
+ *
+ * @param linenum The line number to find
+ * @throws std::runtime_error if line number not found
+ */
+void VVTBI::find_linenum(int linenum) {
+    DEBUG_LOG("Searching for line number: " << linenum);
```

Comment:
This was my naive way to backtrack through the code 10+ years ago, i bet there's a smarter way to have a mapping for `goto` but you can optimize that later